packages/flutter/lib/src/services/text_input.dart

```diff
@@ -135,9 +136,47 @@ class TextInputType {
 
 /// An action the user has requested the text input control to perform.
 enum TextInputAction {
+  /// Show no action button. Android only.
+  none,
+
+  /// Let the OS choose the button. Corresponds to "unspecified" on Android and
+  /// "default" on iOS.
+  unspecified,
+
   /// Complete the text input operation.
   done,
 
+  /// Complete the text input operation. Corresponds to "go" on Android and iOS.
+  go,
+
+  /// Complete the text input operation. Corresponds to "search" on Android and
+  /// iOS.
+  search,
+
+  /// Complete the text input operation. Corresponds to "send" on Android and
+  /// iOS.
+  send,
+
+  /// Complete the text input operation and move focus to the logical "next"
+  /// input. Corresponds to "next" on Android and iOS.
+  next,
+
+  /// Complete the text input operation and move focus to the logical "previous"
+  /// input. Android only.
+  previous,
+
+  /// Complete the text input operation. iOS only, corresponds to "continue."
+  continue_action,
```

Comment:
I think this needs to be camelcase instead of underscore case (like the emergencyCall member below).